tests/ModuleRepositoryTest.php

```diff
@@ -0,0 +1,198 @@
+<?php
+/**
+ * 2007-2019 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Academic Free License (AFL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/AFL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/AFL-3.0  Academic Free License (AFL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+use Symfony\Component\Filesystem\Filesystem;
+use PrestaShop\Module\AutoUpgrade\Module\ModuleRepository;
+use PrestaShop\Module\AutoUpgrade\Module\AddonsClientInterface;
+use PHPUnit\Framework\TestCase;
+
+class ModuleRepositoryTest extends TestCase
+{
+    /** @var string */
+    private $tempDir;
+
+    /** @var string */
+    private $tempModulesDir;
+
+    /** @var string */
+    private $tempDisabledModulesDir;
+
+    /** @var Filesystem */
+    private $fileSystem;
+
+    protected function setUp()
+    {
+        parent::setUp();
+        $this->fileSystem = new Filesystem();
+        $this->tempDir = sys_get_temp_dir() . '/module_disabler';
+        $this->tempModulesDir = $this->tempDir . '/modules';
+        $this->tempDisabledModulesDir = $this->tempDir . '/disabled_modules';
+    }
+
+    private function cleanModules()
```

Comment:
could be in setUp & teardown?

Reply:
Probably, I copied this from `ModuleDisablerTest` where I couldn't because there's a dependency between two tests, but in this case I think I can